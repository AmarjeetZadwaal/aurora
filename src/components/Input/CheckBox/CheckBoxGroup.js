import React from "react";
import PropTypes from "prop-types";
import Provider from "../../SelectionProvider/Provider";
import KeyBoardProvider from "../../KeyboardNavigation/Provider";

const CheckBoxGroup = ({ onChange, children, value }) => (
  <Provider onChange={onChange} value={value} isMultiSelect>
    <KeyBoardProvider role="checkbox">{children}</KeyBoardProvider>
  </Provider>
);

CheckBoxGroup.propTypes = {
  onChange: PropTypes.func,
  children: PropTypes.node,
  value: PropTypes.arrayOf(PropTypes.string)
};

CheckBoxGroup.defaultProps = {
  children: null,
  onChange: null,
  value: []
};

export default CheckBoxGroup;
