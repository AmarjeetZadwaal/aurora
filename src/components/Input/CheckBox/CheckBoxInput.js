import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import colors from "../../../theme/colors";
import constants from "../../../theme/constants";

const CheckBoxInput = styled.input.attrs({
  type: "checkbox"
})`
  outline: none;
  margin: 0;
  appearance: none;
  position: relative;
  .checkbox--large & {
    width: 24px;
    height: 24px;
  }
  .checkbox--small & {
    width: 16px;
    height: 16px;
  }

  &:before {
    content: "";
    background-color: transparent;
    border-radius: ${constants.borderRadius.small};
    border: 1px solid ${colors.onyx.light};
    position: absolute;
    .checkbox--large & {
      width: 24px;
      height: 24px;
    }
    .checkbox--small & {
      width: 16px;
      height: 16px;
    }
    &.checkbox--disabled {
      color: grey;
    }
  }
  &:hover:before {
    border-width: 2px;
    border-color: ${colors.azure.base};
  }
  &:focus:before {
    outline: none;
    .checkbox--small & {
      width: 16px;
      height: 16px;
      border-radius: ${constants.borderRadius.small};
      border: solid 1px ${colors.azure.base};
      background-color: ${colors.white.base};
    }
    .checkbox--large & {
      width: 24px;
      height: 24px;
      border-radius: ${constants.borderRadius.small};
      border: solid 1px ${colors.azure.base};
      background-color: ${colors.white.base};
    }
  }
  &:checked:after {
    content: "";
    position: absolute;
    .checkbox__multi--large & {
      content: "";
      width: 12px;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      height: 3px;
      border-radius: ${constants.borderRadius.small};
      background-color: ${colors.azure.base};
    }
    .checkbox__multi--small & {
      content: "";
      width: 8px;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      height: 2px;
      border-radius: ${constants.borderRadius.small};
      background-color: ${colors.azure.base};
    }
  }
`;

class CheckBoxInputComponent extends React.Component {
  static propTypes = {
    size: PropTypes.oneOf(["large", "small"]),
    isActive: PropTypes.bool,
    name: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    isChecked: PropTypes.bool.isRequired,
    onClick: PropTypes.func.isRequired,
    isFocused: PropTypes.bool.isRequired
  };

  static defaultProps = {
    size: "small",
    isActive: true
  };

  constructor(props) {
    super(props);
    this.SelectedElement = React.createRef();
  }

  componentDidUpdate() {
    this.props.isFocused && this.SelectedElement.current.focus(); // eslint-disable-line
  }

  render() {
    const { name, value, size, isActive, isChecked, ...props } = this.props;

    return (
      <CheckBoxInput
        {...props}
        value={value}
        name={name}
        id={`${name + value}input`}
        disabled={!isActive}
        defaultChecked={isChecked}
        innerRef={this.SelectedElement}
      />
    );
  }
}
export default CheckBoxInputComponent;
