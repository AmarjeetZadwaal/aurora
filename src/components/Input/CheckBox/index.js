/* eslint import/prefer-default-export: off */
export { default as CheckBoxGroup } from "./CheckBoxGroup";
