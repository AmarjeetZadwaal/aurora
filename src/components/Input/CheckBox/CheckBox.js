import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";
import classNames from "classnames";
import Checkmark from "../../Icons/Checkmark";
import { Consumer } from "../../SelectionProvider/Context";
import colors from "../../../theme/colors";
import { Consumer as KeyBoardConsumer } from "../../KeyboardNavigation/Context";
import CheckBoxInput from "./CheckBoxInput";
import composeEventHandlers from "../../../utils/composeEventHandlers";

const CheckBoxWrapper = styled.label`
  position: relative;
  outline: none;
  cursor: pointer;
  display: flex;
  align-items: center;
  margin-bottom: 10px;
  text-align: left;
  color: ${colors.onyx.base};
  &.checkbox--disabled {
    color: grey;
    cursor: default;
    pointer-events: none;
  }
  &.checkbox--large {
    font-size: 16px;
    line-height: 1.5;
  }
  &.checkbox--small {
    font-size: 14px;
    line-height: 1.5;
  }
`;

const CheckBoxText = styled.span`
  display: flex;
  outline: none;
  align-items: center;
  justify-content: center;
  margin-left: 7px;
`;

const CheckDivStyling = styled.div`
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: left;
  position: absolute;
  opacity: 0;

  .checkbox__checked & {
    opacity: 1;
  }
  .checkbox__multi--small & {
    display: none;
  }

  .checkbox__multi--large & {
    display: none;
  }
`;

const StyledCheckmark = styled(Checkmark)`
  margin-top: 4px;
`;

const CheckBoxButton = ({
  name,
  value,
  size,
  isActive,
  children,
  index,
  ...props
}) => (
  <KeyBoardConsumer>
    {({ focused, focusSelected }) => (
      <Consumer>
        {({ selected, onClick }) => {
          const isChecked = selected.includes(value);
          return (
            <CheckBoxWrapper
              className={classNames({
                "checkbox--small": size === "small",
                "checkbox--large": size === "large",
                "checkbox--disabled": !isActive,
                "checkbox__multi--large":
                  selected.length > 1 && size === "large",
                "checkbox__multi--small":
                  selected.length > 1 && size === "small",
                checkbox__checked: isChecked
              })}
            >
              <CheckBoxInput
                {...props}
                value={value}
                name={name}
                id={name + value}
                disabled={!isActive}
                size={size}
                isChecked={isChecked}
                onClick={composeEventHandlers(
                  () => onClick({ value }),
                  () => focusSelected({ index }),
                  props.onClick
                )}
                isFocused={focused === index}
              />
              <CheckDivStyling
                className={classNames({
                  checkbox__checked: isChecked
                })}
              >
                <StyledCheckmark size={size === "small" ? 16 : 24} />
              </CheckDivStyling>
              <CheckBoxText
                role="checkbox"
                aria-labelledby={name + value}
                aria-checked={isChecked}
              >
                {children}
              </CheckBoxText>
            </CheckBoxWrapper>
          );
        }}
      </Consumer>
    )}
  </KeyBoardConsumer>
);

CheckBoxButton.propTypes = {
  size: PropTypes.oneOf(["large", "small"]),
  children: PropTypes.node,
  isActive: PropTypes.bool,
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  index: PropTypes.number.isRequired,
  multiSelect: PropTypes.bool,
  onClick: PropTypes.func
};

CheckBoxButton.defaultProps = {
  size: "small",
  children: null,
  isActive: true,
  multiSelect: false,
  onClick: null
};

export default CheckBoxButton;
