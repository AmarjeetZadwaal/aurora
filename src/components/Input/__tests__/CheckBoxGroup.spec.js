import React from "react";
import { renderIntoDocument, cleanup, fireEvent } from "react-testing-library";

import CheckBoxGroup from "../CheckBox/CheckBoxGroup";
import CheckBox from "../CheckBox/CheckBox";

describe("CheckBoxGroup", () => {
  afterEach(cleanup);

  it("renders default input", () => {
    expect(renderGroupComponent()).toMatchSnapshot();
  });

  it("renders small input", () => {
    expect(renderGroupComponent({ size: "small" })).toMatchSnapshot();
  });

  it("renders large input", () => {
    expect(renderGroupComponent({ size: "large" })).toMatchSnapshot();
  });

  it("onChange receives current value", () => {
    const onChange = jest.fn();

    const { getByTestId } = renderIntoDocument(
      <CheckBoxGroup onChange={onChange}>
        <CheckBox
          name="something"
          data-testid="test-checkbox"
          value="else"
          size="small"
          index={0}
        />
      </CheckBoxGroup>
    );

    fireEvent.click(getByTestId("test-checkbox"));

    expect(onChange).toHaveBeenCalledWith(["else"]);
  });

  it("selecting a single item twice", () => {
    const onChange = jest.fn();

    const { getByTestId } = renderIntoDocument(
      <CheckBoxGroup onChange={onChange}>
        <CheckBox
          name="something"
          data-testid="test-checkbox"
          value="else"
          size="small"
          index={0}
        />
      </CheckBoxGroup>
    );

    fireEvent.click(getByTestId("test-checkbox"));
    fireEvent.click(getByTestId("test-checkbox"));

    expect(onChange).toHaveBeenCalledWith([]);
  });

  it("selecting multiple items", () => {
    const onChange = jest.fn();

    const { getByTestId } = renderIntoDocument(
      <CheckBoxGroup onChange={onChange}>
        <CheckBox
          name="something"
          data-testid="test-checkbox"
          value="1"
          size="small"
          index={0}
        />
        <CheckBox
          name="something"
          data-testid="test-checkbox2"
          value="2"
          size="large"
          index={1}
        />
      </CheckBoxGroup>
    );

    fireEvent.click(getByTestId("test-checkbox"));
    fireEvent.click(getByTestId("test-checkbox2"));

    expect(onChange).toHaveBeenCalledWith(["1", "2"]);
  });

  function renderGroupComponent(props = {}) {
    return renderIntoDocument(
      <CheckBoxGroup {...props}>
        <CheckBox
          name="something"
          data-testid="test-checkbox2"
          value="2"
          size="large"
          index={1}
        />
      </CheckBoxGroup>
    );
  }

  it("passing multiple values", () => {
    const onChange = jest.fn();

    const { getByTestId } = renderIntoDocument(
      <CheckBoxGroup value={["1", "2"]} onChange={onChange}>
        <CheckBox
          name="something"
          data-testid="test-checkbox"
          value="1"
          size="small"
          index={0}
        />
        <CheckBox
          name="something"
          data-testid="test-checkbox2"
          value="2"
          size="large"
          index={1}
        />

        <CheckBox
          name="something"
          data-testid="test-checkbox3"
          value="2"
          size="large"
          index={2}
        />
      </CheckBoxGroup>
    );

    expect(getByTestId("test-checkbox").hasAttribute("checked"));
    expect(getByTestId("test-checkbox2").hasAttribute("checked"));
    expect(!getByTestId("test-checkbox3").hasAttribute("checked"));
  });

  it("passing multiple items then selecting more", () => {
    const onChange = jest.fn();

    const { getByTestId } = renderIntoDocument(
      <CheckBoxGroup value={["1", "2"]} onChange={onChange}>
        <CheckBox
          name="something"
          data-testid="test-checkbox"
          value="1"
          size="small"
          index={0}
        />
        <CheckBox
          name="something"
          data-testid="test-checkbox2"
          value="2"
          size="large"
          index={1}
        />
        <CheckBox
          name="somethingelse"
          data-testid="test-checkbox3"
          value="3"
          size="large"
          index={2}
        />
      </CheckBoxGroup>
    );

    fireEvent.click(getByTestId("test-checkbox3"));

    expect(getByTestId("test-checkbox").hasAttribute("checked"));
    expect(getByTestId("test-checkbox2").hasAttribute("checked"));
    expect(getByTestId("test-checkbox3").hasAttribute("checked"));
  });
});
