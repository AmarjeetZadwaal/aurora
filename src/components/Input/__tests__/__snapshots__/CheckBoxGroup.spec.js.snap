// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxGroup renders default input 1`] = `
.c1 {
  outline: none;
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
}

.checkbox--large .c1 {
  width: 24px;
  height: 24px;
}

.checkbox--small .c1 {
  width: 16px;
  height: 16px;
}

.c1:before {
  content: "";
  background-color: transparent;
  border-radius: 2px;
  border: 1px solid rgba(0,0,15,0.6);
  position: absolute;
}

.checkbox--large .c1:before {
  width: 24px;
  height: 24px;
}

.checkbox--small .c1:before {
  width: 16px;
  height: 16px;
}

.c1:before.checkbox--disabled {
  color: grey;
}

.c1:hover:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:focus:before {
  outline: none;
}

.checkbox--small .c1:focus:before {
  width: 16px;
  height: 16px;
  border-radius: 2px;
  border: solid 1px rgba(2,108,223,1);
  background-color: rgba(255,255,255,1);
}

.checkbox--large .c1:focus:before {
  width: 24px;
  height: 24px;
  border-radius: 2px;
  border: solid 1px rgba(2,108,223,1);
  background-color: rgba(255,255,255,1);
}

.c1:checked:after {
  content: "";
  position: absolute;
}

.checkbox__multi--large .c1:checked:after {
  content: "";
  width: 12px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 3px;
  border-radius: 2px;
  background-color: rgba(2,108,223,1);
}

.checkbox__multi--small .c1:checked:after {
  content: "";
  width: 8px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 2px;
  border-radius: 2px;
  background-color: rgba(2,108,223,1);
}

.c0 {
  position: relative;
  outline: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
  text-align: left;
  color: rgba(0,0,15,1);
}

.c0.checkbox--disabled {
  color: grey;
  cursor: default;
  pointer-events: none;
}

.c0.checkbox--large {
  font-size: 16px;
  line-height: 1.5;
}

.c0.checkbox--small {
  font-size: 14px;
  line-height: 1.5;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 7px;
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  position: absolute;
  opacity: 0;
}

.checkbox__checked .c2 {
  opacity: 1;
}

.checkbox__multi--small .c2 {
  display: none;
}

.checkbox__multi--large .c2 {
  display: none;
}

.c3 {
  margin-top: 4px;
}

<div
  onKeyDown={[Function]}
  role="checkbox"
>
   
  <label
    className="checkbox--large c0"
  >
    <input
      className="c1"
      data-testid="test-checkbox2"
      defaultChecked={false}
      disabled={false}
      id="something2input"
      name="something"
      onClick={[Function]}
      type="checkbox"
      value="2"
    />
    <div
      className="c2"
    >
      <svg
        className="c3"
        fill="none"
        fillRule="evenodd"
        height={24}
        stroke="#026CDF"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={3}
        viewBox="0 0 24 28"
        width={24}
      >
        <path
          d="M5.215 12.219l4.437 4.439 9.514-9.433"
        />
      </svg>
    </div>
    <span
      aria-checked={false}
      aria-labelledby="something2"
      className="c4"
      role="checkbox"
    />
  </label>
</div>
`;

exports[`CheckBoxGroup renders large input 1`] = `
.c1 {
  outline: none;
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
}

.checkbox--large .c1 {
  width: 24px;
  height: 24px;
}

.checkbox--small .c1 {
  width: 16px;
  height: 16px;
}

.c1:before {
  content: "";
  background-color: transparent;
  border-radius: 2px;
  border: 1px solid rgba(0,0,15,0.6);
  position: absolute;
}

.checkbox--large .c1:before {
  width: 24px;
  height: 24px;
}

.checkbox--small .c1:before {
  width: 16px;
  height: 16px;
}

.c1:before.checkbox--disabled {
  color: grey;
}

.c1:hover:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:focus:before {
  outline: none;
}

.checkbox--small .c1:focus:before {
  width: 16px;
  height: 16px;
  border-radius: 2px;
  border: solid 1px rgba(2,108,223,1);
  background-color: rgba(255,255,255,1);
}

.checkbox--large .c1:focus:before {
  width: 24px;
  height: 24px;
  border-radius: 2px;
  border: solid 1px rgba(2,108,223,1);
  background-color: rgba(255,255,255,1);
}

.c1:checked:after {
  content: "";
  position: absolute;
}

.checkbox__multi--large .c1:checked:after {
  content: "";
  width: 12px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 3px;
  border-radius: 2px;
  background-color: rgba(2,108,223,1);
}

.checkbox__multi--small .c1:checked:after {
  content: "";
  width: 8px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 2px;
  border-radius: 2px;
  background-color: rgba(2,108,223,1);
}

.c0 {
  position: relative;
  outline: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
  text-align: left;
  color: rgba(0,0,15,1);
}

.c0.checkbox--disabled {
  color: grey;
  cursor: default;
  pointer-events: none;
}

.c0.checkbox--large {
  font-size: 16px;
  line-height: 1.5;
}

.c0.checkbox--small {
  font-size: 14px;
  line-height: 1.5;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 7px;
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  position: absolute;
  opacity: 0;
}

.checkbox__checked .c2 {
  opacity: 1;
}

.checkbox__multi--small .c2 {
  display: none;
}

.checkbox__multi--large .c2 {
  display: none;
}

.c3 {
  margin-top: 4px;
}

<div
  onKeyDown={[Function]}
  role="checkbox"
>
   
  <label
    className="checkbox--large c0"
  >
    <input
      className="c1"
      data-testid="test-checkbox2"
      defaultChecked={false}
      disabled={false}
      id="something2input"
      name="something"
      onClick={[Function]}
      type="checkbox"
      value="2"
    />
    <div
      className="c2"
    >
      <svg
        className="c3"
        fill="none"
        fillRule="evenodd"
        height={24}
        stroke="#026CDF"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={3}
        viewBox="0 0 24 28"
        width={24}
      >
        <path
          d="M5.215 12.219l4.437 4.439 9.514-9.433"
        />
      </svg>
    </div>
    <span
      aria-checked={false}
      aria-labelledby="something2"
      className="c4"
      role="checkbox"
    />
  </label>
</div>
`;

exports[`CheckBoxGroup renders small input 1`] = `
.c1 {
  outline: none;
  margin: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  position: relative;
}

.checkbox--large .c1 {
  width: 24px;
  height: 24px;
}

.checkbox--small .c1 {
  width: 16px;
  height: 16px;
}

.c1:before {
  content: "";
  background-color: transparent;
  border-radius: 2px;
  border: 1px solid rgba(0,0,15,0.6);
  position: absolute;
}

.checkbox--large .c1:before {
  width: 24px;
  height: 24px;
}

.checkbox--small .c1:before {
  width: 16px;
  height: 16px;
}

.c1:before.checkbox--disabled {
  color: grey;
}

.c1:hover:before {
  border-width: 2px;
  border-color: rgba(2,108,223,1);
}

.c1:focus:before {
  outline: none;
}

.checkbox--small .c1:focus:before {
  width: 16px;
  height: 16px;
  border-radius: 2px;
  border: solid 1px rgba(2,108,223,1);
  background-color: rgba(255,255,255,1);
}

.checkbox--large .c1:focus:before {
  width: 24px;
  height: 24px;
  border-radius: 2px;
  border: solid 1px rgba(2,108,223,1);
  background-color: rgba(255,255,255,1);
}

.c1:checked:after {
  content: "";
  position: absolute;
}

.checkbox__multi--large .c1:checked:after {
  content: "";
  width: 12px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 3px;
  border-radius: 2px;
  background-color: rgba(2,108,223,1);
}

.checkbox__multi--small .c1:checked:after {
  content: "";
  width: 8px;
  top: 50%;
  left: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
  height: 2px;
  border-radius: 2px;
  background-color: rgba(2,108,223,1);
}

.c0 {
  position: relative;
  outline: none;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 10px;
  text-align: left;
  color: rgba(0,0,15,1);
}

.c0.checkbox--disabled {
  color: grey;
  cursor: default;
  pointer-events: none;
}

.c0.checkbox--large {
  font-size: 16px;
  line-height: 1.5;
}

.c0.checkbox--small {
  font-size: 14px;
  line-height: 1.5;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  outline: none;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-left: 7px;
}

.c2 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: left;
  -webkit-justify-content: left;
  -ms-flex-pack: left;
  justify-content: left;
  position: absolute;
  opacity: 0;
}

.checkbox__checked .c2 {
  opacity: 1;
}

.checkbox__multi--small .c2 {
  display: none;
}

.checkbox__multi--large .c2 {
  display: none;
}

.c3 {
  margin-top: 4px;
}

<div
  onKeyDown={[Function]}
  role="checkbox"
>
   
  <label
    className="checkbox--large c0"
  >
    <input
      className="c1"
      data-testid="test-checkbox2"
      defaultChecked={false}
      disabled={false}
      id="something2input"
      name="something"
      onClick={[Function]}
      type="checkbox"
      value="2"
    />
    <div
      className="c2"
    >
      <svg
        className="c3"
        fill="none"
        fillRule="evenodd"
        height={24}
        stroke="#026CDF"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth={3}
        viewBox="0 0 24 28"
        width={24}
      >
        <path
          d="M5.215 12.219l4.437 4.439 9.514-9.433"
        />
      </svg>
    </div>
    <span
      aria-checked={false}
      aria-labelledby="something2"
      className="c4"
      role="checkbox"
    />
  </label>
</div>
`;
